projects/tesseract-ocr/build.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash -eu
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+cd $SRC
+tar zxf leptonica-1.77.0.tar.gz
+cd leptonica-1.77.0
+./configure
+make -j$(nproc)
+make install
+ldconfig
+
+cd $SRC/tesseract
+./autogen.sh
+CXXFLAGS="$CXXFLAGS -D_GLIBCXX_DEBUG" ./configure --disable-graphics --disable-shared
+make -j$(nproc)
+
+cd $SRC/tesseract-ocr-fuzzers
+
+mv $SRC/tessdata $OUT
+export TESSDATA_PREFIX=$(realpath $OUT/tessdata)
```

Comment:
Tesseract requires the ```TESSDATA_PREFIX``` env variable to be set (and point to the directory containing the data files). Because I assumed env variables don't persist across oss-fuzz builds, I hardcoded it into the binary, and set it there upon initialization: https://github.com/guidovranken/tesseract-ocr-fuzzers/blob/master/fuzzer-api.cpp#L47

Is it at all possible to store data files used by the fuzzer binary and if so what path can I use for this?

Reply:
If oss-fuzz doesn't allow data files on disk, then I could pack the required files into the binary, create a ramdisk in ```LLVMFuzzerInitialize``` and write them there, if that is acceptable to you.

Reply:
You can use files on disk.
[This example](https://github.com/google/clusterfuzz/blob/master/docs/setting-up-fuzzing/heartbleed/handshake-fuzzer.cc#L54) is known to work.

Reply:
I think things should work even if you keep the data in a directory (as it is now), but the absolute path cannot be used. Is there any way to make it work with a relative path?

Also, note that `TESSDATA_PREFIX` env variable will not be set on the fuzzing bots. I guess it's ok, since it's being passed during compilation?

Reply:
Is it really necessary to set `TESSDATA_PREFIX`? Tesseract also has a command line option `--tessdata-dir PATH`.

Reply:
The path issue should now be fixed.

@stweil For the fuzzer we don't use the Tesseract binary but create a ```PIX``` struct on the fly with the fuzzer input, and then invoke the API to extract the UTF8 text: https://github.com/guidovranken/tesseract-ocr-fuzzers/blob/master/fuzzer-api.cpp

Reply:
@Dor1s I don't have a seed corpus, but the fuzzer will quickly reach coverage >10000 on its own, and a segfault within minutes.